store/sqlstore/channel_store.go

```diff
@@ -2089,11 +2099,15 @@ func (s SqlChannelStore) UpdateLastViewedAt(channelIds []string, userId string)
 		for _, t := range lastPostAtTimes {
 			times[t.Id] = t.LastPostAt
 		}
+		if updateThreads {
+			go s.Thread().UpdateUnreadsByChannel(userId, threadsToUpdate, now)
```

Comment:
My apologies for an unsolicited comment, I just happened to notice this PR by accident.

I think this has possibility to cause more harm than benefit. I'd rather trade-off slower response times than higher throughput at the expense of increased DB load. We should be a bit careful when spinning up random goroutines without control. If this is a performance critical part that must be asynchronous, I would suggest creating a separate goroutine which receives such commands and batches them and converts them into a single query. That makes for a much controlled workload.

Otherwise, let's make it synchronous.

Same in L2139 too.

Again, sorry to suddenly barge in a review like this.

Reply:
@agnivade - no worries, all reviews are welcome :+1: @jespino thoughts?

Reply:
How does the proposal above ^ change user experience?

Reply:
We need this merged in ASAP, can we try making it synchronous and execute a new performance test?

Reply:
ok, i'll run load tests without goroutines and compare the results

Reply:
I think @streamer45 was already planning to run a 1M test. Claudio - would you be open to running another one by making these calls synchronous? Or let me know I can do that too.

The bigger question is:
- What is the SLA that we are agreeing upon for the delay in marking channels as viewed? From the preliminary tests, it looks like a 2x slowdown. And it will be slightly higher with more channels and more channel members.
- Let's say 2x is acceptable. Now how much more leeway is allowed for it to be slower?

In any case, we should wait before the load tests are done. Following the recent discussion in guild meeting, I would like all performance related concerns to be understood and accepted by everyone before we move on with this.